lighthouse-core/audits/non-composited-animations.js

```diff
@@ -0,0 +1,190 @@
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a diagnostic LH audit that provides details on animations that are not composited. */
+  title: 'Avoid non-composited animations',
+  /** Description of a diagnostic LH audit that shows the user animations that are not composited. */
+  description: 'Animations which are not composited can be janky and contribute to CLS. ' +
+    '[Learn more](https://developers.google.com/web/fundamentals/performance/rendering/stick-to-compositor-only-properties-and-manage-layer-count)',
+  /** [ICU Syntax] Label identifying the number of animations that are not composited. */
+  displayValue: `{itemCount, plural,
+  =1 {# animated element found}
+  other {# animated elements found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * Each failure reason is represented by a bit flag. The bit shift operator '<<' is used to define which bit corresponds to each failure reason.
+ * https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/animation/compositor_animations.h;l=58;drc=f959aba6099c5d767f885b7f4632955b463dc41d?originalUrl=https:%2F%2Fcs.chromium.org%2F
+ * @type {{flag: number, text: string}[]}
+ */
+const ACTIONABLE_FAILURE_REASONS = [
+  {
+    flag: 1 << 13,
+    text: 'Unsupported CSS Property',
+  },
+];
+
+/**
+ * Return list of actionable failure reasons and a boolean if some reasons are not actionable.
+ * Each flag is a number with a single bit set to 1 in the position corresponding to a failure reason.
+ * We can check if a specific bit is true in the failure coding using bitwise and '&' with the flag.
+ * @param {number} failureCode
+ * @return {string[]}
+ */
+function getActionableFailureReasons(failureCode) {
+  /** @type {string[]} */
+  const failureReasons = [];
+  ACTIONABLE_FAILURE_REASONS.forEach(reason => {
+    if (failureCode & reason.flag) {
+      failureReasons.push(reason.text);
+    }
+  });
+  return failureReasons;
+}
+
+class NonCompositedAnimations extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'non-composited-animations',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'TraceElements', 'HostUserAgent'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    // This audit reqiures m86
+    const match = artifacts.HostUserAgent.match(/Chrome\/(\d+)/);
+    if (!match || Number(match[1]) < 86) {
+      return {
+        score: 1,
+        notApplicable: true,
+      };
+    }
+
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const traceOfTab = await TraceOfTab.request(trace, context);
```

Comment:
seems strange that we still need to pull in trace-of-tab for this here, should we just expose `compositeFailed` on each animation entry in trace elements gatherer instead since we're already chasing down those trace events there?

Reply:
Good point, this code is leftover from before we added the gatherer.